README.md

```diff
@@ -0,0 +1,496 @@
+
+The first word of the comment should be capitalized, unless the first word is an identifier that begins with a lower-case letter.
+
+If a comment is short, the period at the end can be omitted.
+
+<a name="block_comments"></a>
+### Block Comments
+
+Block comments apply to the block of code that follows them.
+
+Each line of a block comment starts with a `#` and a single space, and should be indented at the same level of the code that it describes.
+
+Paragraphs inside of block comments are separated by a line containing a single `#`.
+
+```coffeescript
+  # This is a block comment. Note that if this were a real block
+  # comment, we would actually be describing the proceeding code.
+  #
+  # This is the second paragraph of the same block comment. Note
+  # that this paragraph was separated from the previous paragraph
+  # by a line containing a single comment character.
+
+  init()
+  start()
+  stop()
+```
+
+<a name="inline_comments"></a>
+### Inline Comments
+
+Inline comments are placed on the line immediately above the statement that they are describing. If the inline comment is sufficiently short, it can be placed on the same line as the statement (separated by a single space from the end of the statement).
+
+All inline comments should start with a `#` and a single space.
+
+The use of inline comments should be limited, because their existence is typically a sign of a code smell.
+
+Do not use inline comments when they state the obvious:
+
+```coffeescript
+  # No
+  x = x + 1 # Increment x
+```
+
+However, inline comments can be useful in certain scenarios:
+
+```coffeescript
+  # Yes
+  x = x + 1 # Compensate for border
+```
+
+<a name="naming_conventions"></a>
+## Naming Conventions
+
+Use `camelCase` (with a leading lowercase character) to name all variables, methods, and object properties.
+
+Use `CamelCase` (with a leading uppercase character) to name all classes. _(This style is also commonly referred to as `PascalCase`, `CamelCaps`, or `CapWords`, among [other alternatives][camel-case-variations].)_
+
+_(The **official** CoffeeScript convention is camelcase, because this simplifies interoperability with JavaScript. For more on this decision, see [here][coffeescript-issue-425].)_
+
+For constants, use all uppercase with underscores:
+
+```coffeescript
+CONSTANT_LIKE_THIS
+```
+
+Methods and variables that are intended to be "private" should begin with a leading underscore:
+
+```coffeescript
+_privateMethod: ->
+```
+
+<a name="functions"></a>
+## Functions
+
+_(These guidelines also apply to the methods of a class.)_
+
+When declaring a function that takes arguments, always use a single space after the closing parenthesis of the arguments list:
+
+```coffeescript
+foo = (arg1, arg2) -> # Yes
+foo = (arg1, arg2)-> # No
+```
+
+Do not use parentheses when declaring functions that take no arguments:
+
+```coffeescript
+bar = -> # Yes
+bar = () -> # No
```

Comment:
I'm inclined to agree. It gets even more confusing when you use a fat arrow, e.g. `bar = =>`.